src/net_processing.cpp

```diff
@@ -4219,6 +4222,49 @@ void PeerManagerImpl::MaybeSendAddr(CNode& node, std::chrono::microseconds curre
     }
 }
 
+void PeerManagerImpl::MaybeSendFeefilter(CNode& pto, std::chrono::microseconds current_time)
+{
+    AssertLockHeld(cs_main);
+
+    if (m_ignore_incoming_txs) return;
+    if (!pto.m_tx_relay) return;
+    if (pto.GetCommonVersion() < FEEFILTER_VERSION) return;
+    // peers with the forcerelay permission should not filter txs to us
+    if (pto.HasPermission(NetPermissionFlags::ForceRelay)) return;
+
+    CAmount currentFilter = m_mempool.GetMinFee(gArgs.GetArg("-maxmempool", DEFAULT_MAX_MEMPOOL_SIZE) * 1000000).GetFeePerK();
+    static FeeFilterRounder g_filter_rounder{CFeeRate{DEFAULT_MIN_RELAY_TX_FEE}};
+
+    if (m_chainman.ActiveChainstate().IsInitialBlockDownload()) {
+        // Received tx-inv messages are discarded when the active
+        // chainstate is in IBD, so tell the peer to not send them.
+        currentFilter = MAX_MONEY;
+    } else {
+        static const CAmount MAX_FILTER{g_filter_rounder.round(MAX_MONEY)};
```

Comment:
Thanks for the input - this further improves the readability

Reply:
`FeeFilterRounder::round()` is not a pure function - it calls to `FastRandomContext::rand32()` and so will return different values even when called with the same input. Therefore this is a change in behaviour.

In general, we shouldn't mix behaviour changes with refactors in the same PR. It may be the right change, but it should be left for a separate PR, leaving this PR as a pure refactor.

Reply:
@jnewbery It is not clear to me that the behavior changes--in both cases `round()` is only called once.

Reply:
Interesting -- thanks @jnewbery.

Reply:
Thanks @jnewbery for catching that. Removed that commit.